repos/system_upgrade/el7toel8/actors/checkkdegnome/libraries/library.py

```diff
@@ -0,0 +1,158 @@
+    with open(path, "r") as f:
+        for line in f.readlines():
+            if "xsession" in line.lower():
+                default_xsession = line.split("=")[1].lower()
+    return default_xsession
+
+
+def check_app_in_use(app):
+    """
+    Method return True if application was used in last month, False in other cases.
+    """
+    path = "{0}/.kde/share/config/{1}rc".format(os.environ.get("HOME"), app)
+    if os.path.isfile(path):
+        last_modified = os.stat(path).st_mtime
+        # Application is considered actively used, if it has been used in last month.
+        return last_modified >= int(time.time() - ONE_MONTH)
+    return False
+
+
+def is_installed(app):
+    """
+    Wrapper for "rpm -q <app>" command, returns True if application is found,
+    False in other cases. Output of rpm command is not supressed.
+    """
+    return True if subprocess.call(["rpm", "-q", app]) == 0 else False
+
+
+def check_kde_gnome():
+    apps_in_use = 0
+    api.current_logger().info("  Detecting desktop environments  ")
+    api.current_logger().info("==================================")
+
+    # Detect installed desktops by their startup files
+    kde_desktop_installed = is_executable("/usr/bin/startkde")
+    gnome_desktop_installed = is_executable("/usr/bin/gnome-session")
+    api.current_logger().info("* KDE installed: {0}".format(kde_desktop_installed))
+    api.current_logger().info("* Gnome installed: {0}".format(gnome_desktop_installed))
+    api.current_logger().info("----------------------------------")
+
+    # No desktop installed, we don't even care about apps as they are most likely not used or even installed
+    if not kde_desktop_installed and not gnome_desktop_installed:
+        api.current_logger().info("No desktop installed. Continuing with the upgrade.")
+        return
+
+    if kde_desktop_installed:
+        api.current_logger().info("KDE desktop is installed. Checking what we can do about it.")
+        if not gnome_desktop_installed:
+            api.current_logger().error("Cannot perform the upgrade because there is\
+             no other desktop than KDE installed.")
+            # We cannot continue with the upgrade process
+            reporting.create_report([
+                reporting.Title("Cannot upgrade because there is no other desktop than KDE installed."),
+                reporting.Summary("With only KDE installed, there would be no other desktop after upgrade."),
+                reporting.Severity(reporting.Severity.HIGH),
+                reporting.Tags([
+                    reporting.Tags.UPGRADE_PROCESS
+                ]),
+                reporting.Flags([
+                    reporting.Flags.INHIBITOR
+                ]),
+                reporting.Remediation(
+                    hint="Install GNOME desktop to be able to upgrade.")
+                ])
+            return
+
+        # Assume GNOME is installed in this state
+
+        user = os.environ.get("USER")
```

Comment:
TODO: @pirat89 : check ipuconfig

Reply:
Same issue as above. You should not rely on os.environ in the checksphase. From this point, we already provide set of `LEAPP_DEVEL` env vars in the Actor.configuration object. Not sure, whether we should extend this to get all variables or whether will be better to create actor that will provide information about the rest of variables. @vinzenz @fernflower @oamg/developers  - WDYT guys? I believe that definitely we should not create new actor that provide information just about one, two, specific envvars, but to have something really more generic. Not sure what we should propose as solution here.

Reply:
In this case, also HOME envar does not make any sense.  I will try to make a workaround.